src/script/InitDelegateeDeposits.s.sol

```diff
@@ -0,0 +1,152 @@
+import {stdJson} from "forge-std/StdJson.sol";
+import {Multicall} from "@openzeppelin/contracts/utils/Multicall.sol";
+import {IGovLst, Staker} from "../interfaces/IGovLst.sol";
+
+/// @title InitDelegateeDeposits
+/// @author [ScopeLift](https://scopelift.co)
+/// @notice A script to initialize deposits for delegatee addresses in the GovLst contract.
+/// @dev This script reads delegatee addresses from a JSON file, checks if they already have deposits,
+/// and initializes deposits for those that don't have one yet using multicall batching for efficiency.
+abstract contract InitDelegateeDeposits is Script {
+  /// @notice Reference to the GovLst contract.
+  IGovLst govLst = IGovLst(getGovLstAddress());
+
+  /// @notice The default deposit ID value used to identify uninitialized deposits.
+  uint256 public immutable DEFAULT_DEPOSIT_ID = Staker.DepositIdentifier.unwrap(govLst.DEFAULT_DEPOSIT_ID());
+
+  /// @notice The number of operations to include in each multicall batch.
+  uint256 public immutable BATCH_SIZE = multicallBatchSize();
+
+  /// @notice Main entry point for the script.
+  /// @dev Reads addresses from file, filters those needing initialization, and processes them in batches.
+  function run() public virtual {
+    vm.startBroadcast();
+
+    address[] memory _delegateeAddresses = readDelegateeAddressesFromFile();
+    uint256[] memory _depositIds = getDepositIdsForDelegateeAddresses(_delegateeAddresses);
+    (address[] memory _addressesToInit, uint256 _numOfAddressesToInit) =
+      filterDelegateeAddresses(_delegateeAddresses, _depositIds);
+    (, uint256 _batchCount) = callFetchOrInitializeDepositForDelegatee(_addressesToInit, _numOfAddressesToInit);
+
+    console2.log("\n========== INITIALIZATION SUMMARY ==========");
+    console2.log("Total delegatee addresses processed:", _delegateeAddresses.length);
+    console2.log("Addresses filtered (already have deposits):", _delegateeAddresses.length - _numOfAddressesToInit);
+    console2.log("Addresses included for initialization:", _numOfAddressesToInit);
+    console2.log("Multicall batch size:", BATCH_SIZE);
+    console2.log("Number of multicall batches submitted:", _batchCount);
+    console2.log("===========================================");
+
+    vm.stopBroadcast();
+  }
+
+  /// @notice Returns the address of the GovLst contract.
+  /// @return The address of the GovLst contract.
+  function getGovLstAddress() public virtual returns (address);
+
+  /// @notice Returns the batch size for multicall operations.
+  /// @return The number of operations to include in each batch.
+  function multicallBatchSize() public pure virtual returns (uint256);
+
+  /// @notice Reads delegatee addresses from a JSON file.
+  /// @return An array of delegatee addresses.
+  function readDelegateeAddressesFromFile() public view virtual returns (address[] memory) {
+    string memory _root = vm.projectRoot();
+    string memory _path = string.concat(_root, "/src/script/addresses.json");
```

Comment:
Rather than forcing the inheritor of this contract to use a hardcoded address path, I think we should include a virtual method for the address file path. The natspec for that virtual method should include instructions to add the file to the foundry.toml file and also give a brief explanation of the expected keys/format.

Reply:
Added `filePath` function that can be overriden with natspec of how to do it. https://github.com/withtally/stGOV/pull/91/commits/d571a2c8e9ce9d6dc7dcdcaec724cc8c30831f47

Reply:
Might just want to add a quick note that the file should have a top level array consisting of a list of addresses as strings, or something along these lines. Otherwise looks good.

Reply:
https://github.com/withtally/stGOV/pull/91/commits/121cd02d6780d6b7b1ca401c8ada9df696e0d858